wcfsetup/install/files/lib/system/event/listener/AbstractDynamicDatabaseObjectEventListener.class.php

```diff
@@ -0,0 +1,58 @@
+<?php
+namespace wcf\system\event\listener;
+use wcf\data\AbstractDatabaseObjectAction;
+
+/**
+ * Special event listener for `\wcf\data\AbstractDatabaseObjectAction` events.
+ * This appends behind the called function name the action of the database object
+ *
+ * Using:
+ *      - Event Name = finalizeAction
+ *      - \wcf\data\AbstractDatabaseObjectAction::$action = create
+ *      =>  ::onFinalizeActionCreate($eventObj, array &$parameters)
+ *
+ * @author       Olaf Braun
+ * @copyright    2001-2019 WoltLab GmbH
+ * @license      GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @package      WoltLabSuite\Core\System\Event\Listener
+ */
+abstract class AbstractDynamicDatabaseObjectEventListener extends AbstractDynamicInvocationEventListener {
+	
+	/**
+	 * @var AbstractDatabaseObjectAction
+	 */
+	protected $eventObj;
+	
+	/**
+	 * @param AbstractDatabaseObjectAction $eventObj
+	 * @param array                        $parameters
+	 */
+	protected function onFinalizeAction($eventObj, &$parameters) {
```

Comment:
Can you do
```suggestion
	protected function onFinalizeAction($eventObj, array &$parameters) {
```
or does this cause issues for some reason?